components/eventing-controller/pkg/backend/nats/jetstream/jetstream_test.go

```diff
@@ -1476,3 +1477,103 @@ func TestSubscriptionSubjectIdentifierNamespacedName(t *testing.T) {
 		})
 	}
 }
+
+// TestJetStream_NoNATSSubscription tests if the error is being triggered
+// when expected in-memory representation of NATS subscriptions are missing.
+func TestJetStream_NoNATSSubscription(t *testing.T) {
+	// given
+	testEnvironment := setupTestEnvironment(t)
+	jsBackend := testEnvironment.jsBackend
+	defer testEnvironment.natsServer.Shutdown()
+	defer testEnvironment.jsClient.natsConn.Close()
+	initErr := jsBackend.Initialize(nil)
+	require.NoError(t, initErr)
+
+	// create New Subscriber
+	subscriber := evtesting.NewSubscriber()
+	defer subscriber.Shutdown()
+	require.True(t, subscriber.IsRunning())
+
+	testCases := []struct {
+		name                            string
+		subOpts                         []evtesting.SubscriptionOpt
+		givenManuallyDeleteSubscription bool
+		givenFilterToDelete             string
+		wantNatsSubsLen                 int
+		wantErr                         bool
+		wantErrText                     string
```

Comment:
suggest making this *string so you can set it to nil

Reply:
This creates some other problem, when I want to assign some values to it. So I would leave it as is. If I leave the `wantErr` flag, I don't need to check the `wantErrText` for nil. Instead I always know when I expect an error and when not.

Reply:
It introduces some problems when I want to assign some concatenated strings to it. IMO we can leave it as is. We need it to be nil if we have a flag `wantErr` which tells us when we expect an err and to define which one.